### PR TITLE
Update spring-boot version to 3.1.0

### PR DESCRIPTION
**Pull request by DanielFran**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mraible** [comment]
I tried checking out this branch and running `npm run update-snapshots` followed by `npm test`. There are still test failures:

```
  8599 passing (1m)
  386 pending
  2 failing

  1) generator - sql - database-url
       getR2dbcUrl
         when called for mariadb
           return r2dbc:mysql://localhost:3306/test?useLegacyDatetimeCode=false&serverTimezone=UTC:
     Error: expect(received).toEqual(expected) // deep equality

Expected: "r2dbc:mysql://localhost:3306/test?useLegacyDatetimeCode=false&serverTimezone=UTC"
Received: "r2dbc:mariadb://localhost:3306/test?useLegacyDatetimeCode=false&serverTimezone=UTC"
      at Context.<anonymous> (generators/spring-data-relational/support/database-url.spec.mts:149:87)
      at process.processImmediate (node:internal/timers:471:21)

  2) generator - sql - database-url
       getR2dbcUrl
         when called for mariadb with skipExtraOptions enabled
           return r2dbc:mysql://localhost:3306/test:
     Error: expect(received).toEqual(expected) // deep equality

Expected: "r2dbc:mysql://localhost:3306/test"
Received: "r2dbc:mariadb://localhost:3306/test"
      at Context.<anonymous> (generators/spring-data-relational/support/database-url.spec.mts:156:111)
      at process.processImmediate (node:internal/timers:471:21)
```

I'm not sure if the code should be fixed to return MySQL or if the test should be fixed to expect MariaDB.

**Comment by mraible** [comment]
We should probably fix this too:

```
Base64Utils in org.springframework.util has been deprecated and marked for removal
```

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/Base64Utils.html

**Comment by mraible** [comment]
I added a bug bounty for this. I don't think we need it for v8 beta, but I think we should have it for v8 GA.

**Comment by BrayanMnz** [comment]
What is missing here for the spring boot upgrade? @mraible @DanielFran

**Comment by mraible** [comment]
If you look at the failing CI jobs, you'll see many are failing. Combing through the release notes might reveal why. I'm not sure.

https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.1-Release-Notes

**Comment by mshima** [comment]
@DanielFran may I help? Would be nice to have this for beta.2.

**Comment by DanielFran** [comment]
@mshima yes go head.
I have very few time, only on weekends...

**Comment by mshima** [comment]
Fixed:
- r2dbc-mariadb driver still fails in backend test at GitHub. Works locally like reactive MongoDB. So disabled backend test at sample.
Keeping r2dbc-mysql for mariadb is not a good choice too, since MariaDB v11 fails with it. see https://github.com/jhipster/generator-jhipster/pull/22502.
- Environment v4 integration have a misbehavior generating workspaces, fix deployed to yeoman-environment 4.0.0-beta.1 and adjusts to bootstrap generator.
- Incorporate mongodb database config into url config, looks a bug at spring-boot, but workaround is easy.
- Adjusts to AuthorizationHeaderUtilTest.

Todo:
- HibernateTimeZoneIT is still failing locally with a different timezone.

File: generators/generator-constants.mts [hunk 1]
```diff
@@ -32,26 +32,23 @@ export const javaDependencies: Record<string, string> = {
   /**
    * spring-boot version should match the one managed by https://mvnrepository.com/artifact/tech.jhipster/jhipster-dependencies/JHIPSTER_DEPENDENCIES_VERSION
    */
-  'spring-boot': '3.0.7',
+  'spring-boot': '3.1.0',
   /*
    * hibernate version should match the one managed by https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/SPRING_BOOT_VERSION
    * Required due to hibernate-jpamodelgen annotation processor.
    */
+  // TODO drop fixed 6.1.7.Final version. Refer to https://github.com/jhipster/generator-jhipster/issues/22579
```
Comment (mshima) [inline generators/generator-constants.mts:40]:
```suggestion
  // TODO update to spring-boot provided version. Refer to https://github.com/jhipster/generator-jhipster/issues/22579
```

**Comment by mshima** [comment]
I think it’s ready.

**Comment by mshima** [comment]
@DanielFran lets split the bounty of this PR and https://github.com/jhipster/generator-jhipster/pull/22933.
I will claim this one you claim the other one.

**Comment by mshima** [comment]
@DanielFran bounty claimed https://opencollective.com/generator-jhipster/expenses/170971.

**Comment by DanielFran** [comment]
@mshima approved

**Pull request merged**